test/unit/org/apache/cassandra/auth/PasswordAuthenticatorTest.java

```diff
@@ -198,4 +200,11 @@ public void warmCacheWithEmptyTable()
         Map<String, String> cacheEntries = authenticator.bulkLoader().get();
         assertTrue(cacheEntries.isEmpty());
     }
+
+    @Test
+    public void testDefaultAuthenticateMessage()
+    {
+        AuthenticateMessage authenticateMessage = authenticator.getAuthenticateMessage(null, null);
+        assertThat(authenticateMessage.authenticator).isEqualTo(PasswordAuthenticator.class.getName());
+    }
```

Comment:
```suggestion
    @Test
    public void testDefaultAuthenticateMessage()
    {
        AuthenticateMessage authenticateMessage = authenticator.getAuthenticateMessage(null, null);
        assertThat(authenticateMessage.authenticator).isEqualTo(PasswordAuthenticator.class.getName());
    }
    
    @Test
    public void testDefaultSimpleNameAuthenticateMessage()
    {
        AuthenticateMessage authenticateMessage = authenticator.getAuthenticateMessage(null, null);
        assertThat(authenticateMessage.authenticator).isEqualTo(PasswordAuthenticator.class.getSimpleName());
    }
    
    @Test
    public void testCustomAuthenticateMessage()
    {
        AuthenticateMessage authenticateMessage = authenticator.getAuthenticateMessage(null, null);
        assertThat(authenticateMessage.authenticator).isEqualTo(CustomTestPasswordAuthenticator.class.getName());
    }
    
    public static class CustomTestPasswordAuthenticator extends PasswordAuthenticator {}
```
…?

Reply:
The second test, using `PasswordAuthenticator.class.getSimpleName()` is not accurate.
The fact that we're able to use a short-form in the configuration, internally we should always send the FQCN (which is also what the drivers expect).

The third test is valuable. If we strictly required such test, we should do it outside of the `PasswordAuthenticatorTest` as this is functionality not related with `PasswordAuthenticator` but with `IAuthenticator` implementations in general (we could have a simpler unit test instead of `CQLTester`).